parser/src/lib.rs

```diff
@@ -259,13 +272,46 @@ fn parse_document_blocks(input: &str) -> IResult<&str, Vec<Ast>> {
     preceded(
         many0(line_ending),
         separated_list0(
-            preceded(line_ending, many1(line_ending)),
+            many1(line_ending),
             map(module::parse_multiline_module, Ast::Module)
+                .or(map(parse_heading, Ast::Heading))
                 .or(map(parse_paragraph, Ast::Paragraph)),
         ),
     )(input)
 }
 
+/// Parses a heading which consists of a sequence of hashtags to indicate heading level
+/// followed by text. The level and text are put into a `Heading` node.
+///
+/// # Arguments
+///
+/// * `input`: The text to parse
+///
+/// returns: The heading node, if a successful parse occurs, otherwise the parse error
+fn parse_heading(input: &str) -> IResult<&str, Heading> {
+    map(
+        pair(
+            verify(take_while1(|c| c == '#'), |s: &str| s.len() <= 6),
```

Comment:
I think we should allow an arbitrary amount of levels for headings (limited of course by the data type used, let's say an u8) and then each "Heading" transforms can be responsible for the how to handle the level.

This would avoid being too HTML centric, in LaTeX for instance there might not be a meaningful interpretation of Heading 6 since it stops at `\subsubsection{...}`. And some other presentational format might find it handy to support even more levels.